File: tus-client/src/shared/entity/quest.ts
import { Wellbeing } from './wellbeing';
import { Item } from './item';
import { EntitySerializable } from './entity-serializable';
import { Entity } from './entity';


export class Quest extends Entity implements EntitySerializable<Quest> {

    questNo: number;
    finished: boolean;
    created: Date;
    updated: Date;
    slots: Item[];
    wbs: Wellbeing[];

    deserialize(input: any): Quest {
        this.questNo = input.questNo;
        this.finished = input.finished;
        this.created = new Date(input.created);
        this.updated = new Date(input.updated);

        if (input.slots != undefined) {
            this.slots = new Array<Item>();
            for (const slot of input.slots) {
                this.slots.push(new Item().deserialize(slot));
            }
        }

        if (input.wbs != undefined) {
            this.wbs = new Array<Wellbeing>();
            for (const wb of input.wbs) {
                this.wbs.push(new Wellbeing().deserialize(wb));
            }
        }


        return this;
    }
}

File: tus-server/src/main/java/at/statistik/tus/entity/Quest.java
package at.statistik.tus.entity;

import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true, exclude = { "user" })
@Entity
@Table(name = "TQUEST")
public class Quest extends EntityTus {

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "USER_ID", columnDefinition = "BIGINT", nullable = false)
	@JsonIgnore
	private User user;

	@JsonProperty
	public Long getUserId() {
		return this.user == null ? null : this.user.getId();
	}

	@Column(name = "QUEST_NO")
	private Integer questNo;

	@Column(name = "FINISHED")
	private Boolean finished;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "CREATED")
	private Date created;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "UPDATED")
	private Date updated;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "quest", cascade = { CascadeType.DETACH, CascadeType.MERGE,
			CascadeType.PERSIST, CascadeType.REFRESH })
	private List<Slot> slots;

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "quest", cascade = { CascadeType.DETACH, CascadeType.MERGE,
			CascadeType.PERSIST, CascadeType.REFRESH })
	private List<Wb> wbs;

}
